auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -267,9 +257,7 @@ private void registerUser(final String email, final String name, final String pa
                 .addOnSuccessListener(getActivity(), new OnSuccessListener<AuthResult>() {
                     @Override
                     public void onSuccess(AuthResult authResult) {
-                        SaveSmartLock saveSmartLock = getSaveSmartLock();
```

Comment:
Here's the fatal flaw that let us catch the issue: as you'll see below, in the other spots we do this, it works because we initialize the fragment _before_ starting a sign-in. Thus, the fragment has time to do its thing by the time it's actually used (since network calls are slow). However, in this case, we immediately use the fragment below and go BOOM! 😁

The solution used in this PR: add a lifecycle listener that kicks off the process in `onStart` when we are guaranteed to have an activity. Doing that fixes #856!